HW28/HW28.cpp

```diff
@@ -0,0 +1,160 @@
+﻿
+#include <iostream>
+#include <vector>
+#include <thread>
+#include <mutex>
+
+std::vector<int> createVector(std::size_t count)
+{
+	std::vector<int> vec(count);
+
+	for (int i = 0; i < count; ++i) {
+		vec[i] = i % 11; // Fill with values ​​from 0 to 10
+	}
+	return vec;
+}
+
+std::mutex locer;
+void calculateSum(const std::vector<int>& vec, int startIndex, int lastIndex, long& sum)
+{
+	for (int i = startIndex; i <= lastIndex; ++i)
+	{
+		locer.lock();
+		sum += vec[i];
+		locer.unlock();
```

Comment:
using mutex for EVERY addition is highly inefficient. You can see that in your results as well: The more threads you have more inefficient it gets.
addition is much faster compared to lock, so here and in general locks should be used as minimally as possible.

You need to make sure addition is synchronized properly without lock here